pkg/webhooks/validation/controller/controller.go

```diff
@@ -329,12 +329,18 @@ func (c *Controller) reconcileRequest(req reconcileRequest) (bool, error) {
 		// no point in retrying unless cert file changes.
 		return false, nil
 	}
-	failurePolicy := kubeApiAdmission.Ignore
-	ready := c.readyForFailClose()
-	if ready {
+	failurePolicy := kubeApiAdmission.Fail
+	// Only cheeck for fail open if webhooks are not already configured for fail.
+	for i := range configuration.Webhooks {
+		if configuration.Webhooks[i].FailurePolicy != nil && *configuration.Webhooks[i].FailurePolicy == kubeApiAdmission.Ignore {
+			failurePolicy = kubeApiAdmission.Ignore
+			break
+		}
+	}
+	if failurePolicy != kubeApiAdmission.Fail && c.readyForFailClose() {
```

Comment:
I think current code allows changing from fail back to ignore. Are we intentionally removing that?

What is the motivation for this pr?

Reply:
> I think current code allows changing from fail back to ignore. 

I thought it is only doing one way i.e. ignore -> fail . that is why I changed it like that. Let me take a look again.